tests/lib/openstack_api/test_openstack_service.py

```diff
@@ -0,0 +1,106 @@
+from unittest.mock import MagicMock, NonCallableMagicMock
+from openstack_api.openstack_service import disable_service, enable_service
+
+
+def test_disable_service_function_disables_service_when_enabled():
+    """tests that openstack disables the service when it is currently enabled"""
+
+    # mock params
+    mock_conn = MagicMock()
+    mock_service = MagicMock()
+    mock_hypervisor_name = NonCallableMagicMock()
+    mock_service_binary = NonCallableMagicMock()
+    mock_disable_reason = NonCallableMagicMock()
+
+    mock_service.status = "enabled"
+
+    # run the function
+    res = disable_service(
+        mock_conn,
+        mock_service,
+        mock_hypervisor_name,
+        mock_service_binary,
+        mock_disable_reason,
+    )
+
+    # assert expectations
+    # assert conn.compute.disable_service was called appropriately
+    mock_conn.compute.disable_service.assert_called_once_with(
+        mock_service,
+        host=mock_hypervisor_name,
+        binary=mock_service_binary,
+        disabled_reason=mock_disable_reason,
+    )
+    # test that function returns == conn.compute.disable_service.return_value
+    assert res == mock_conn.compute.disable_service.return_value
+
+
+def test_disable_service_function_returns_none_when_disabled():
```

Comment:
See above comment about naming. In this case I would append `fails` or something similar to the end to give a rough idea. But leave the details to be in the description.
```suggestion
def test_disable_service_fails():
```